src/main/java/net/fabricmc/loom/api/LoomGradleExtensionAPI.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of fabric-loom, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.loom.api;
+
+import java.io.File;
+import java.util.List;
+
+import org.gradle.api.Action;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.file.ConfigurableFileCollection;
+
+import net.fabricmc.loom.api.decompilers.LoomDecompiler;
+import net.fabricmc.loom.configuration.ide.RunConfigSettings;
+import net.fabricmc.loom.configuration.processors.JarProcessor;
+import net.fabricmc.loom.configuration.providers.mappings.LayeredMappingSpecBuilder;
+
+/**
+ * This is the public api available exposed to build scripts.
+ */
+public interface LoomGradleExtensionAPI {
+	File getAccessWidener();
+
+	void setAccessWidener(Object file);
+
+	void setShareCaches(boolean shareCaches);
+
+	default void shareCaches() {
+		setShareCaches(true);
+	}
+
+	List<LoomDecompiler> getDecompilers();
+
+	void addDecompiler(LoomDecompiler decompiler);
+
+	List<JarProcessor> getJarProcessors();
+
+	void addJarProcessor(JarProcessor processor);
+
+	ConfigurableFileCollection getLog4jConfigs();
+
+	default Dependency officialMojangMappings() {
+		return layered(LayeredMappingSpecBuilder::officialMojangMappings);
+	}
+
+	Dependency layered(Action<LayeredMappingSpecBuilder> action);
+
+	String getRefmapName();
+
+	void setRefmapName(String refmapName);
+
+	boolean isRemapMod();
+
+	void setRemapMod(boolean remapMod);
```

Comment:
Note that moving from a plain `remapMod` field to a getter/setter pair like this is a breaking change for Kotlin buildscripts (the generated property name is now `isRemapMod`), but it probably doesn't hurt:
- most people don't use this
- most people don't use kts buildscripts
- 0.9 is unstable in the first place

Reply:
I think this is fine, mostly becuase I dont know of any case where this is even used.

I expect there might be stuff that is missed from this, we can easily evaulate it and add it back if needed.